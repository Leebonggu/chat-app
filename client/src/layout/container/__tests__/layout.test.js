import React from 'react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import Layout from '../layout';

const mockStore = configureStore([thunk]);
jest.mock('../../../lib/components/Grid', () => ({ Grid: 'Grid' }));
jest.mock('../footer.version2.js', () => 'Footer');
jest.mock('../header', () => 'Header');

test('Layout Container should render given props', () => {
  const initialState = {};
  const store = mockStore(initialState);
  const component = renderer.create(
    <Provider store={store}>
      <Layout>Test</Layout>
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
