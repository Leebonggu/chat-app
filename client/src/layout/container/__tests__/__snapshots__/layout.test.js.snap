// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Container should render given props 1`] = `
<div
  className="layout-container"
>
  <Header />
  <div
    className="layout-content"
  >
    <Grid>
      Test
    </Grid>
  </div>
  <Footer />
</div>
`;
